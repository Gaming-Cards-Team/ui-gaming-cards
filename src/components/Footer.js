import React from "react";
import Navbar from "react-bootstrap/Navbar";
import "bootstrap/dist/css/bootstrap.min.css";
import { FOOTER_TEXT } from "../util/constant.js"

const textStyle={
    color: "white",
    width: "100%",
    textAlign: "center",
  };
  const footerStyle={
    backgroundColor: "#2d963f"
  };

const Footer = () => (
  <Navbar 
  fixed="bottom"
  style={footerStyle}
  >
    <div
      data-testid="footer-text"
      style={textStyle}
    >
      {FOOTER_TEXT}
    </div>
  </Navbar>
);

export default Footer;
