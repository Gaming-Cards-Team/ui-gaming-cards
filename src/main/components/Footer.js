import React from "react";
import Navbar from "react-bootstrap/Navbar";
import "bootstrap/dist/css/bootstrap.min.css";
import { FOOTER_TEXT } from "../util/constant.js"

const textStyle={
    color: "white",
    width: "100%",
    textAlign: "center",
  };

const Footer = () => (
  <Navbar bg="success" fixed="bottom">
    <div
      data-testid="footer-text"
      style={textStyle}
    >
      {FOOTER_TEXT}
    </div>
  </Navbar>
);

export default Footer;
