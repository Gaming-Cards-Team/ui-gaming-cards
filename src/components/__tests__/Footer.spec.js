import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Footer from '../Footer';

describe('Test Footer component', () => {

    it('should have a fixed text', () => {
        const { getByTestId } = render(<Footer />);
        const footerText = getByTestId('footer-text')

        expect (footerText.textContent).toBe('Copyright Star Wars and all associated names and/or images are copyright Lucasfilm Ltd.')
    });

    it('should validate the component', () => { 
        const footer = renderer.create(<Footer />).toJSON();
        
        expect(footer).toMatchSnapshot();
    });

});