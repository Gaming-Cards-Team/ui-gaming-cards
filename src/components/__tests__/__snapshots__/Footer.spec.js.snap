// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer component should validate the component 1`] = `
<nav
  className="navbar navbar-expand navbar-light fixed-bottom"
  style={
    Object {
      "backgroundColor": "#2d963f",
    }
  }
>
  <div
    data-testid="footer-text"
    style={
      Object {
        "color": "white",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    Copyright Star Wars and all associated names and/or images are copyright Lucasfilm Ltd.
  </div>
</nav>
`;
